roles/kubernetes-apps/csi_driver/cinder/templates/cinder-csi-nodeplugin.yml.j2

```diff
@@ -121,3 +121,5 @@ spec:
             path: {{ kube_config_dir }}/cinder-cacert.pem
             type: FileOrCreate
 {% endif %}
+      tolerations:
+        {{ cinder_tolerations | to_nice_yaml(indent=2) | indent(width=8) }}
```

Comment:
I hesitate to add empty tolerations into the spec.
For sample, `ingress_nginx_tolerations` is checked when adding it as https://github.com/kubernetes-sigs/kubespray/blob/1c3d082b8dce3e05fd4f905b4e15798071d3d97f/roles/kubernetes-apps/ingress_controller/ingress_nginx/templates/ds-ingress-nginx-controller.yml.j2#L33-L36

It is better to add the similar condition here like

```
{% if cinder_tolerations %}
      tolerations:
        {{ cinder_tolerations | to_nice_yaml(indent=2) | indent(width=8) }}
{% endif %}

Reply:
Thanks for the feedback.

I pushed this change as a separate commit - do you want me to squash the commits myself or will you squash them when you merge?

Reply:
Thank you for updating the pull request.
It is better to squash multiple commits into a single by yourself before merging.
In addition, please rebase the pull request based on the latest master branch to solve molecule_tests issue.

Sorry for this inconvenience.

Reply:
No problem. I squashed the commits and rebased on master, hope everything looks good now.